### PR TITLE
[SPARK-40142][PYTHON][SQL][FOLLOW-UP] Make pyspark.sql.functions examples self-contained (part 7, ~30 functions)

### PR DESCRIPTION
**Pull request by khalidmammadov**
### What changes were proposed in this pull request?
It's part of the Pyspark docstrings improvement series (https://github.com/apache/spark/pull/37592, https://github.com/apache/spark/pull/37662, https://github.com/apache/spark/pull/37686, https://github.com/apache/spark/pull/37786, https://github.com/apache/spark/pull/37797)

In this PR I mainly covered missing parts in the docstrings adding some more examples where it needed.

### Why are the changes needed?
To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?
Yes, documentation

### How was this patch tested?
```
PYTHON_EXECUTABLE=python3.9 ./dev/lint-python
./python/run-tests --testnames pyspark.sql.functions
bundle exec jekyll build
```

### CODE REVIEW

**Comment by khalidmammadov** [comment]
@HyukjinKwon @srowen @itholic  Please review

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -6108,6 +6297,11 @@ def schema_of_json(json: "ColumnOrName", options: Optional[Dict[str, str]] = Non
         .. versionchanged:: 3.0
            It accepts `options` parameter to control schema inferring.
 
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        a string representatio of a :class:`StructType` parsed from given JSON.
```
Comment (srowen) [inline python/pyspark/sql/functions.py:6303]: typo in a few places: representatio -> representation

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -5797,6 +5915,11 @@ def array_except(col1: "ColumnOrName", col2: "ColumnOrName") -> Column:
     col2 : :class:`~pyspark.sql.Column` or str
         name of column containing array
 
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        an array of values from first array that is not in the second.
```
Comment (srowen) [inline python/pyspark/sql/functions.py:5921]: nit: that are not

File: python/pyspark/sql/functions.py [hunk 3]
```diff
@@ -5535,6 +5570,20 @@ def array_join(
 
     .. versionadded:: 2.4.0
 
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+        target column to work on.
+    delimiter : str
+        delimiter to use concatanate elements
```
Comment (srowen) [inline python/pyspark/sql/functions.py:5578]: concatenate, here and below
Also: "delimiter used to concatenate..."

**Review by srowen — CHANGES_REQUESTED** [review]

File: python/pyspark/sql/functions.py [hunk 4]
```diff
@@ -5619,6 +5708,10 @@ def element_at(col: "ColumnOrName", extraction: Any) -> Column:
     >>> df = spark.createDataFrame([(["a", "b", "c"],)], ['data'])
     >>> df.select(element_at(df.data, 1)).collect()
     [Row(element_at(data, 1)='a')]
+    >>> df.select(element_at(df.data, -1)).collect()
+    [Row(element_at(data, -1)='c')]
+    >>> df.select(element_at(df.data, -4)).collect()
+    [Row(element_at(data, -4)=None)]
```
Comment (itholic) [inline python/pyspark/sql/functions.py:5714]: Can we add a short description why this returns `None`?

e.g.

```
    >>> df.select(element_at(df.data, -1)).collect()
    [Row(element_at(data, -1)='c')]

    Returns `None` if there is no value corresponding to the given `extraction`.

    >>> df.select(element_at(df.data, -4)).collect()
    [Row(element_at(data, -4)=None)]
```

File: python/pyspark/sql/functions.py [hunk 5]
```diff
@@ -6144,6 +6338,11 @@ def schema_of_csv(csv: "ColumnOrName", options: Optional[Dict[str, str]] = None)
 
         .. # noqa
 
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        a string representatio of a :class:`StructType` parsed from given CSV.
```
Comment (itholic) [inline python/pyspark/sql/functions.py:6344]: > typo in a few places: representatio -> representation

Here, too

**Review by itholic — COMMENTED** [review]
Looks fine otherwise

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**